packages/amplify-graphql-sql-transformer/src/graphql-sql-transformer.ts

```
@@ -0,0 +1,148 @@
+          );
+        }
+
+        const statement = config.statement ?? (config.reference ? context.customQueries.get(config.reference) : MISSING_QUERY_STATEMENT);
+        const resolverResourceId = ResolverResourceIDs.ResolverResourceID(config.resolverTypeName, config.resolverFieldName);
+        const resolver = context.resolvers.generateQueryResolver(
+          config.resolverTypeName,
+          config.resolverFieldName,
+          resolverResourceId,
+          dataSource as any,
+          MappingTemplate.s3MappingTemplateFromString(
+            generateSqlLambdaRequestTemplate(statement ?? MISSING_QUERY_STATEMENT, 'RAW_SQL', config.resolverFieldName),
+            `${config.resolverTypeName}.${config.resolverFieldName}.req.vtl`,
+          ),
+          MappingTemplate.s3MappingTemplateFromString(
+            generateSqlLambdaResponseMappingTemplate(),
+            `${config.resolverTypeName}.${config.resolverFieldName}.res.vtl`,
+          ),
+        );
+
+        resolver.setScope(context.stackManager.getScopeFor(resolverResourceId, SQL_DIRECTIVE_STACK));
+        context.resolvers.addResolver(config.resolverTypeName, config.resolverFieldName, resolver);
+      });
+    });
+  };
+}
+
+export const generateSqlLambdaRequestTemplate = (statement: string, operation: string, operationName: string): string => {
+  return printBlock('Invoke RDS Lambda data source')(
+    compoundExpression([
+      set(ref('lambdaInput'), obj({})),
+      set(ref('lambdaInput.parameters'), obj({})),
+      set(ref('lambdaInput.statement'), str(statement)),
+      set(ref('lambdaInput.operation'), str(operation)),
+      set(ref('lambdaInput.operationName'), str(operationName)),
+      set(ref('lambdaInput.parameters'), methodCall(ref('util.defaultIfNull'), ref('context.arguments'), obj({}))),
+      obj({
```

Is there a clean way to detect and fail fast if the variables expected by the SQL statement do not exist in the input? This will save customers invalid lambda invocations.